test_conformance/subgroups/test_subgroup_ballot.cpp

```diff
@@ -78,25 +126,54 @@ template <typename Ty> struct BALLOT
                 {
                     current_sbs = wg_offset + sbs > lws ? lws - wg_offset : sbs;
                 }
-                for (wi_id = 0; wi_id < current_sbs; ++wi_id)
+
+                bs128 expected_result_bs = 0;
+
+                std::set<int> active_work_items;
+                for (int wi_id = 0; wi_id < current_sbs; ++wi_id)
                 {
-                    device_result = my[wg_offset + wi_id];
-                    expected_result = 1;
-                    if (!compare(device_result, expected_result))
+                    if (test_params.work_items_mask.test(wi_id))
+                    {
+                        bool predicate = (mx[wg_offset + wi_id].s0 != 0);
+                        expected_result_bs |= (bs128(predicate) << wi_id);
+                        active_work_items.insert(wi_id);
+                    }
+                }
+                if (active_work_items.empty())
+                {
+                    continue;
```

Comment:
It might be worth pointing out why this is being skipped.
Given that the following loop is iterating over the contents of the container, imediatelyfollowed the loop restart, perhaps this condition is not necessary.

Reply:
I see now this is not necessary because in next lines we start iterating over `active_work_items` so in case it is empty we skip running checking results.

Reply:
It's certainly not necessary but I think the structure of this still makes some sense. If there are no active work items, there is no expected result, and it makes sense not to enter the scope containing a variable for such a result.